lib/workers/global/autodiscover.ts

```diff
@@ -80,3 +75,24 @@ export async function autodiscoverRepositories(
   }
   return { ...config, repositories: discovered };
 }
+
+export function applyFilters(repos: string[], filters: string[]): string[] {
+  const matched = new Set<string>();
+
+  for (const filter of filters) {
+    let res: string[];
+    if (isConfigRegex(filter)) {
+      const autodiscoveryPred = configRegexPredicate(filter);
+      if (!autodiscoveryPred) {
+        throw new Error(`Failed to parse regex pattern "${filter}"`);
+      }
+      res = repos.filter(autodiscoveryPred);
+    } else {
+      res = repos.filter(minimatch.filter(filter));
+    }
+    for (const repository of res) {
+      matched.add(repository);
+    }
```

Comment:
I think we neeed to remove repo from `matched` when a negated regex is in use? eg 

```js
[ 'my-group/**', '!/my-group\\/my-sub-group\\/.*/']
```
This won't work as expected in
- #16466

Reply:
I have planned to introduce a `autodiscoverFilterExclude`, which will remove entries after the fact, when this is merged.      
This hadn't been added here to prevent this PR become to bloated, but we can add it together if it is fine for you.

From a UX perpesctive I think it is better to not give the question mark a special meaning which differs from other parts of Renovate.

Reply:
Ok, but we need to document, that the negative regex don't work as expected when multiple filters are used .

Reply:
Now added to documentation. Would be nice if we can include this as part of v33 though